cli/src/diff_util.rs

```diff
@@ -749,6 +759,109 @@ pub fn show_diff_summary(
+) -> DiffStat {
+    let hunks = unified_diff_hunks(left_content, right_content, 0);
+    let mut adds = 0;
+    let mut dels = 0;
+    for hunk in hunks {
+        for (line_type, _content) in hunk.lines {
+            match line_type {
+                DiffLineType::Context => {}
+                DiffLineType::Removed => dels += 1,
+                DiffLineType::Added => adds += 1,
+            }
+        }
+    }
+    DiffStat {
+        path: repo_path.clone(),
+        added: adds,
+        removed: dels,
+    }
+}
+
+pub fn show_diff_stat(
+    formatter: &mut dyn Formatter,
+    workspace_command: &WorkspaceCommandHelper,
+    tree_diff: TreeDiffIterator,
+) -> Result<(), CommandError> {
+    let mut stats: Vec<DiffStat> = vec![];
+    let mut max_path_length = 0;
+    let mut max_diffs = 0;
+    for (repo_path, diff) in tree_diff {
+        match diff {
+            tree::Diff::Modified(left, right) => {
+                let left_content = diff_content(workspace_command.repo(), &repo_path, &left)?;
+                let right_content = diff_content(workspace_command.repo(), &repo_path, &right)?;
+                let stat = get_diff_stat(&repo_path, &left_content, &right_content);
+                stats.push(stat);
+            }
+            tree::Diff::Added(right) => {
+                let right_content = diff_content(workspace_command.repo(), &repo_path, &right)?;
+                let stat = get_diff_stat(&repo_path, &[], &right_content);
+                stats.push(stat);
+            }
+            tree::Diff::Removed(left) => {
+                let left_content = diff_content(workspace_command.repo(), &repo_path, &left)?;
+                let stat = get_diff_stat(&repo_path, &left_content, &[]);
+                stats.push(stat);
+            }
+        }
+        max_path_length = max(max_path_length, repo_path.to_internal_file_string().len());
+        max_diffs = stats
+            .iter()
+            .map(|stat| max(max_diffs, stat.added + stat.removed))
+            .max()
+            .unwrap_or(0);
+    }
+
+    // TODO: Figure out how to get the terminal width and removed the hardcoded 120.
+    let max_bar_length = 120 - max_path_length - " | ".len() - max_diffs.to_string().len() - 1;
+    let factor = if max_diffs < max_bar_length {
+        1.0
+    } else {
+        max_bar_length as f64 / max_diffs as f64
+    };
+
+    formatter.with_label("diff", |formatter| {
+        let mut total_added = 0;
+        let mut total_removed = 0;
+        for stat in &stats {
+            total_added += stat.added;
+            total_removed += stat.removed;
+            let bar_added = (stat.added as f64 * factor).round() as usize;
+            let bar_removed = (stat.removed as f64 * factor).round() as usize;
+            // pad to max_path_length
+            write!(
+                formatter.labeled("hunk-header"),
+                " {:<max_path_length$} | {:>4}",
+                workspace_command.format_file_path(&stat.path),
```

Comment:
It's probably best to save this formatted path instead of the `RepoPath` in `DiffStat`, so the math based on its length is accurate (it's a relative path, so matters if you're in a subdirectory).

Reply:
Good point. Fixed.